src/BinSkim.Rules/PERules/BA6002.EliminateDuplicateStrings.cs

```diff
@@ -0,0 +1,147 @@
+
+            reasonForNotAnalyzing = MetadataConditions.ImageIsResourceOnlyBinary;
+            if (portableExecutable.IsResourceOnly) { return result; }
+
+            reasonForNotAnalyzing = MetadataConditions.CouldNotLoadPdb;
+            if (target.Pdb == null) { return result; }
+
+            reasonForNotAnalyzing = MetadataConditions.ImageIsNotBuiltWithMSVC;
+            if (!portableExecutable.IsTargetCompiledWithMSVC(target.Pdb)) { return result; }
+
+            reasonForNotAnalyzing = null;
+            return AnalysisApplicability.ApplicableToSpecifiedTarget;
+        }
+
+        public override void AnalyzePortableExecutableAndPdb(BinaryAnalyzerContext context)
+        {
+            PEBinary target = context.PEBinary();
+            Pdb pdb = target.Pdb;
+
+            var compilandsBinaryWithoutStringPooling = new List<ObjectModuleDetails>();
+            var compilandsLibraryWithoutStringPooling = new List<ObjectModuleDetails>();
+
+            foreach (DisposableEnumerableView<Symbol> omView in pdb.CreateObjectModuleIterator())
+            {
+                Symbol om = omView.Value;
+                ObjectModuleDetails omDetails = om.GetObjectModuleDetails();
+
+                if (omDetails.Language != Language.C &&
+                    omDetails.Language != Language.Cxx &&
+                    omDetails.Language != Language.MASM)
+                {
+                    continue;
+                }
+
+                if (!omDetails.HasDebugInfo)
+                {
+                    continue;
+                }
+
+                bool isMSVC = (omDetails.WellKnownCompiler == WellKnownCompilers.MicrosoftC ||
+                               omDetails.WellKnownCompiler == WellKnownCompilers.MicrosoftCxx);
+                if (isMSVC)
+                {
+                    if (!omDetails.EliminateDuplicateStringsEnabled)
+                    {
+                        CompilandRecord record = om.CreateCompilandRecord();
+                        if (!string.IsNullOrEmpty(record.Library))
+                        {
+                            compilandsLibraryWithoutStringPooling.Add(omDetails);
+                        }
+                        else
+                        {
+                            compilandsBinaryWithoutStringPooling.Add(omDetails);
+                        }
+                    }
+                }
+            }
+
+            if (compilandsLibraryWithoutStringPooling.Count > 0 || compilandsBinaryWithoutStringPooling.Count > 0)
+            {
+                if (compilandsLibraryWithoutStringPooling.Count > 0)
+                {
+                    GenerateCompilandsAndLog(context, compilandsLibraryWithoutStringPooling);
+                }
+
+                if (compilandsBinaryWithoutStringPooling.Count > 0)
+                {
+                    GenerateCompilandsAndLog(context, compilandsBinaryWithoutStringPooling);
+                }
+
+                return;
+            }
```

Comment:
Is there a reason to nest these if statements? The outer `if` doesn't check anything additional to the inner ones.

Reply:
I referenced a pre-existing rule that calls out objs/libs by name and it had this same logic (BA2004).  The return statement does prevent a Pass from being logged so this does seem to be meaningful.  The if blocks are independent so if/elseif/else is not appropriate.